modules/web/.husky/pre-commit

```diff
@@ -16,5 +16,12 @@
 . "$(dirname "$0")/_/husky.sh"
 cd modules/web/.husky
 
+# If there are no changes in modules/web, exit the hook early so we don't run on BE-only changes.
+# Since we cd into husky dir, run git diff based on git root directory
+if git diff --quiet main -- "$(git rev-parse --show-toplevel)/modules/web" ; then
```

Comment:
`main` is not required here.

Reply:
To quote myself from the description

> There is a really weird reason though why we need to compare with main: Running git diff --quiet on a branch that has only new files staged, fails because new files cannot be diffed against even if they just have been added
Here's a full example:

```sh
❯ touch modules/web/file
❯ git add modules/web/file
❯ git status
# On branch: limit-husky-to-fe  |  [*] => $e*
#
➤ Changes to be committed
#
#       new file:  [1] modules/web/file
#
❯ git diff --quiet -- "$(git rev-parse --show-toplevel)/modules/web" # will succeed, even though it should fail
❯ git diff --quiet main -- "$(git rev-parse --show-toplevel)/modules/web" # fails as expected
```

Reply:
Yeah but that reasoning is not valid. Those stages are staged so they'll only appear when you do something like `git diff --cached`. TLDR in the `pre-commit` hook you can simply do something like :

```
. "$(dirname "$0")/_/husky.sh"
cd modules/web/.husky

# If there are no changes in modules/web, exit the hook early so we don't run on BE-only changes.
# Since we cd into husky dir, run git diff based on git root directory
web_dir_path=$(git rev-parse --show-toplevel)/modules/web

if git diff --quiet --name-only -- $web_dir_path || git diff --quiet --name-only --cached -- $web_dir_path
then
  npx lint-staged
  npm run check
else
  exit 0
fi

```

Reply:
oh nice, this does the trick!